lib/BackgroundJob/Tasks/CheckCronTask.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * @copyright Copyright (c) 2017, Matias De lellis <mati86dl@gmail.com>
+ * @copyright Copyright (c) 2018, Branko Kokanovic <branko@kokanovic.org>
+ *
+ * @author Branko Kokanovic <branko@kokanovic.org>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+namespace OCA\FaceRecognition\BackgroundJob\Tasks;
+
+use OCA\FaceRecognition\BackgroundJob\FaceRecognitionBackgroundTask;
+use OCA\FaceRecognition\BackgroundJob\FaceRecognitionContext;
+
+/**
+ * Check that we are started either through command, or from cron/webcron (but do not allow ajax mode)
+ */
+class CheckCronTask extends FaceRecognitionBackgroundTask {
+	/**
+	 * @inheritdoc
+	 */
+	public function description() {
+		return "Check that service is started from either cron or from command";
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function do(FaceRecognitionContext $context) {
+		$this->setContext($context);
+
+		$isCommand = $context->isRunningThroughCommand();
```

Comment:
This is not always true?
You set that in https://github.com/matiasdelellis/facerecognition/pull/33/files#diff-3eec1bf8e625d99e338e741f9c524b9cR83

Reply:
So, I am keeping default value as `false` since (once we have background service), it will not be possible to set this in background service (background service class will be `BackgroundService.php`. So, this means that, in command, we have opportunity to set this to true and I am doing that in [line 83 here](https://github.com/matiasdelellis/facerecognition/pull/33/files#diff-3eec1bf8e625d99e338e741f9c524b9cR83). This is how we differentiate between command and service (this is my thinking).

Reply:
Ok.. ;)